src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -105,5 +135,30 @@ private bool CanWrite(PropertyInfo propertyInfo)
             // CanWrite gets true even if a property has only a private setter
             return propertyInfo.CanWrite && (propertyInfo.SetMethod?.IsPublic ?? false);
         }
+
+        private PropertyInfo[] GetProperties(TypeInfo typeInfo)
+        {
+            var propertyBindingsFlags = BindingFlags.Public | BindingFlags.Instance;
+
+            return
+                typeInfo.GetProperties(propertyBindingsFlags)
+                .Where(p => p.GetCustomAttribute<BsonIgnoreAttribute>() == null)
```

Comment:
I don't like this change since we split attributes processing from one convention into at least two. But it's the easiest way, each other good looking way which I tried required much more changes. But I would consider more good looking refactoring for this case in the scope of a new ticket.
Also, it looks like a similar change should be considered for other conventions, at least for `ReadWriteMemberFinderConvention`

Reply:
So the general pattern of the conventions working together is:

1. IClassMapConvention(s) add or remove members from the class map
2. IMemberMapConventions(s) configure member maps added in step 1
3. ICreateMapConvention(s) configure any creator maps added in step 1
4. IPostProcessingConvention(s) do any clean up on the class map

Why don't you try limiting the processing this convention does in the IClassMapConvention to only adding members to the class map, and postpone adding the constructor to the class map until the IPostProcessingConvention. That way the AttributeConventionPack will get a change to remove the `[BsonIgnore]` elements this class added because it doesn't (or shouldn't) know about the `[BsonIgnore]` attribute.

We don't usually add creators in the post processing step, but that might be what's needed here?

Reply:
> Also, it looks like a similar change should be considered for other conventions, at least for ReadWriteMemberFinderConvention

I don't think so. The attribute conventions run after the `ReadWriteMemberFinderConvention` and will remove the members tagged with `[BsonIgnore]`.

Reply:
>Why don't you try limiting the processing this convention does in the IClassMapConvention to only adding members to the class map, and postpone adding the constructor to the class map until the IPostProcessingConvention. That way the AttributeConventionPack will get a change to remove the [BsonIgnore] elements this class added because it doesn't (or shouldn't) know about the [BsonIgnore] attribute.

because of `NamedParameterCreatorMapConvention` which is called here: https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Bson/Serialization/Conventions/ConventionRunner.cs#L69. This convention configures missed arguments for ctor and it requires having already configured ctorMap

Reply:
>I don't think so. The attribute conventions run after the ReadWriteMemberFinderConvention and will remove the members tagged with [BsonIgnore]

agreed